modules/xmpp/moderator.js

```diff
@@ -335,28 +335,31 @@ export default class Moderator extends Listenable {
                                         && text.indexOf('400 invalid-session') > 0;
                                     const notAuthorized = response.status === 403;
 
-                                    this._handleError(roomJid, sessionError, notAuthorized, resolve);
+                                    this._handleError(roomJid, sessionError, notAuthorized, resolve, reject);
                                 })
                                 .catch(error => {
                                     logger.warn(`Error: ${error}`);
-                                    this._handleError(roomJid);
+                                    this._handleError(roomJid, undefined, undefined, resolve, reject);
                                 });
 
                             // _handleError has either scheduled a retry or fired an event indicating failure.
                             return;
                         }
                         response.json()
                             .then(resultJson => {
-                                this._handleSuccess(roomJid, resultJson, resolve);
+                                this._handleSuccess(roomJid, resultJson, resolve, reject);
                             });
                     })
                     .catch(error => {
                         logger.warn(`Error: ${error}`);
-                        this._handleError(roomJid);
+                        this._handleError(roomJid, undefined, undefined, resolve, reject);
                     });
             }
         }).then(() => {
             this.conferenceRequestSent = true;
+        })
+        .finally(() => {
```

Comment:
Would be great if you could add catch -> log some error and then add then to proceed with this. In general I would suggest to avoid using finally as mush as possible as it does not work right away in RN